router/relay/relay.go

```diff
@@ -1158,15 +1158,15 @@ func (rst *RelayStateImpl) PrepareRelayStep(cmngr poolmgr.PoolMgr) error {
 	case nil:
 		return nil
 	case ErrSkipQuery:
-		if err := rst.Client().ReplyErrMsg(err.Error()); err != nil {
+		if err := rst.Client().ReplyErr(err); err != nil {
 			return err
 		}
 		return ErrSkipQuery
 	case qrouter.MatchShardError:
-		_ = rst.Client().ReplyErrMsg("failed to match any datashard")
+		_ = rst.Client().ReplyErrMsg("failed to match any datashard", "SPQRD")
```

Comment:
we can write this as simple as rst.Client().ReplyErrMsgByCode(SPQR_NO_DATASCPACE) (define SPQR_NO_DATASCPACE const)